### PR TITLE
Add option for processing using various size of data chunks in `evp_test`

### PR DESCRIPTION
**Pull request by xhanulik**
This PR implements processing test input data using various sizes of data chunks in `evp_test` as discussed in  #21424.

The required data size is passed via the command line option. The data chunk processing is implemented for cipher, digest, digest sign a digest verify, encoding and MAC tests. When no chunk size is set or set to `0`, the default size (mostly entire data) is processed.

### CODE REVIEW

**Comment by beldmit** [comment]
ready for review

**Comment by beldmit** [comment]
ping @openssl/committers to review

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by beldmit** [comment]
Ping @openssl/committers for the 2nd review

File: test/evp_test.c [hunk 1]
```diff
@@ -88,6 +89,8 @@ static KEY_LIST *public_keys;
 
 static int find_key(EVP_PKEY **ppk, const char *name, KEY_LIST *lst);
 static int parse_bin(const char *value, unsigned char **buf, size_t *buflen);
+static int parse_bin_chunk(const char *value, size_t offset, size_t max,
+                        unsigned char **buf, size_t *buflen, size_t *out_offset);
```
Comment (paulidale) [inline test/evp_test.c:93]: Looks like an indentation nit here.

File: test/evp_test.c [hunk 2]
```diff
@@ -148,23 +152,42 @@ static void evp_test_buffer_free(EVP_TEST_BUFFER *db)
...
+        } else {
+            /* parse in chunks */
+            size_t new_offset = 0;
 
-    if (*sk == NULL && !TEST_ptr(*sk = sk_EVP_TEST_BUFFER_new_null()))
-        goto err;
-    if (!sk_EVP_TEST_BUFFER_push(*sk, db))
-        goto err;
+            if ((rv = parse_bin_chunk(value, offset, max_len, &db->buf,
+                                 &db->buflen, &new_offset)) == -1)
```
Comment (paulidale) [inline test/evp_test.c:177]: indentation nit here too

File: test/evp_test.c [hunk 3]
```diff
@@ -338,6 +361,66 @@ static int parse_bin(const char *value, unsigned char **buf, size_t *buflen)
     return 1;
 }
 
+/*
+ * Convert at maximum "max" bytes to a binary allocated buffer.
+ * Return 1 on success, -1 on failure or 0 for end of value string.
+ */
+static int parse_bin_chunk(const char *value, size_t offset, size_t max,
+                     unsigned char **buf, size_t *buflen, size_t *out_offset)
```
Comment (paulidale) [inline test/evp_test.c:369]: indentation nit.

**Review by paulidale — APPROVED** [review]
Check the indentation nits which should be addressed.

**Comment by beldmit** [comment]
I reconfirm, no need to reset the timer
@paulidale could you please reconfirm?

**Comment by paulidale** [comment]
Reconfirmed.  No need to reset timers.

**Comment by paulidale** [comment]
Merged, thanks for the contribution.

**Pull request closed without merging**